### PR TITLE
MGDAPI-3805 - gosec: address G601

### PR DESCRIPTION
**Pull request by KevFan**
# Issue link

https://issues.redhat.com/browse/MGDAPI-3805
# What

Address gosec G601 - https://stackoverflow.com/questions/62446118/implicit-memory-aliasing-in-for-loop

# Verification steps

* Passing gosec prow check 
* Passing e2e should be sufficient

### CODE REVIEW

File: controllers/rhmi/rhmi_controller.go [hunk 1]
```diff
@@ -1180,24 +1182,24 @@ func (r *RHMIReconciler) processStage(installation *rhmiv1alpha1.RHMI, stage *St
 		}
 
 		// Verify that watches for this productStatus CRDs have been created
-		config, err := configManager.ReadProduct(productStatus.Name)
+		productConfig, err := configManager.ReadProduct(productName)
```
Comment (adam-cattermole) [inline controllers/rhmi/rhmi_controller.go:1185]: minor nit - if we change the config read to use `productName` we should probably adjust these err messages below to use the key also, or we should change this back to retrieving from `productStatus.Name`

Comment (KevFan) [reply controllers/rhmi/rhmi_controller.go:1185]: Good catch !  Will revert this back to `productStatus.Name` :+1:

**Comment by adam-cattermole** [comment]
Thanks for the change

/lgtm

**Comment by KevFan** [comment]
/test rhoam-e2e

@adam-cattermole Rebased against master to resolve conflicts if you can have a look again

**Comment by adam-cattermole** [comment]
Sure, all still looks great

/lgtm

**Pull request merged**